From ffc35dd5bde5cdd4117bc9a6b29c9d6678179cc8 Mon Sep 17 00:00:00 2001
From: Meng Li <Meng.Li@windriver.com>
Date: Wed, 5 Nov 2025 15:35:15 +0800
Subject: [PATCH] arm64: dts: enable cgroup memory controller for raspberrypi5
 platform

The current default boot arguments include cgroup_disable=memory. this
setting was introduced by commit 3a44d1c ("BCM2708: Add core Device Tree
support").

However, the Raspberry Pi 5 has sufficient memory to support the cgroup
memory controller, This functionality was supported in previous versions
without being explicitly disabled, so its current disabling appears to be
mistaken.

Remove cgroup_disable=memory from the boot arguments to leave the cgroup
memory controller enabled by default.

Upstream-Status: Pending

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/broadcom/bcm2712-rpi.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/broadcom/bcm2712-rpi.dtsi b/arch/arm64/boot/dts/broadcom/bcm2712-rpi.dtsi
index 2f6e4a569b3e..c8c3cff982dc 100644
--- a/arch/arm64/boot/dts/broadcom/bcm2712-rpi.dtsi
+++ b/arch/arm64/boot/dts/broadcom/bcm2712-rpi.dtsi
@@ -116,7 +116,7 @@ pio: &rp1_pio {
 
 / {
 	chosen: chosen {
-		bootargs = "reboot=w coherent_pool=1M 8250.nr_uarts=1 pci=pcie_bus_safe cgroup_disable=memory numa_policy=interleave nvme.max_host_mem_size_mb=0";
+		bootargs = "reboot=w coherent_pool=1M 8250.nr_uarts=1 pci=pcie_bus_safe numa_policy=interleave nvme.max_host_mem_size_mb=0";
 		stdout-path = "serial10:115200n8";
 	};
 
-- 
2.34.1

